tests/lib/rules/no-deprecated-slot-attribute.js

```diff
@@ -0,0 +1,287 @@
+      <template>
+        <LinkList>
+          <a v-slot:name />
+        </LinkList>
+      </template>`,
+      errors: [
+        {
+          message: '`slot` attributes are deprecated.',
+          line: 4
+        }
+      ]
+    },
+    {
+      code: `
+      <template>
+        <LinkList>
+          <a slot="name" disabled slot-scope="{a}"/>
+        </LinkList>
+      </template>`,
+      output: `
+      <template>
+        <LinkList>
+          <a v-slot:name="{a}" disabled />
+        </LinkList>
+      </template>`,
+      errors: [
+        {
+          message: '`slot` attributes are deprecated.',
+          line: 4
+        }
+      ]
+    },
+    {
+      code: `
+      <template>
+        <LinkList>
+          <template slot="name" scope="{a}">
+            <a />
+          </template>
+        </LinkList>
+      </template>`,
+      output: `
+      <template>
+        <LinkList>
+          <template v-slot:name="{a}" >
+            <a />
+          </template>
+        </LinkList>
+      </template>`,
+      errors: [
+        {
+          message: '`slot` attributes are deprecated.',
+          line: 4
+        }
+      ]
+    },
+    {
+      code: `
+      <template>
+        <LinkList>
+          <a slot="nameFoo" />
+        </LinkList>
+      </template>`,
+      output: `
+      <template>
+        <LinkList>
+          <a v-slot:nameFoo />
+        </LinkList>
+      </template>`,
+      errors: [
+        {
+          message: '`slot` attributes are deprecated.',
+          line: 4
+        }
+      ]
+    },
+    {
+      code: `
+      <template>
+        <LinkList>
+          <a slot="f o o" />
+          <a slot="obj.prop" />
+          <a slot="a/b" />
+          <a slot="a=b" />
+          <a slot="a>b" />
+        </LinkList>
+      </template>`,
+      output: null,
+      errors: [
+        {
+          message: '`slot` attributes are deprecated.',
+          line: 4
+        },
+        {
+          message: '`slot` attributes are deprecated.',
+          line: 5
+        },
+        {
+          message: '`slot` attributes are deprecated.',
+          line: 6
+        },
+        {
+          message: '`slot` attributes are deprecated.',
+          line: 7
+        },
+        {
+          message: '`slot` attributes are deprecated.',
+          line: 8
+        }
+      ]
+    },
+    {
+      code: `
+      <template>
+        <LinkList>
+          <a v-bind:slot=name />
+        </LinkList>
+      </template>`,
+      output: `
+      <template>
+        <LinkList>
+          <a v-slot:[name] />
+        </LinkList>
+      </template>`,
+      errors: [
+        {
+          message: '`slot` attributes are deprecated.',
+          line: 4
+        }
+      ]
+    },
+    {
+      code: `
+      <template>
+        <LinkList>
+          <a :slot="slot.name" />
+        </LinkList>
+      </template>`,
+      output: `
+      <template>
+        <LinkList>
+          <a v-slot:[slot.name] />
```

Comment:
Actually, this case is invalid because dot in dynamic argument causes syntax error.

```
{
    "directive": true,
    "name": { "name": "slot" },
    "argument": { "name": "[slot" },
    "modifiers": [{ "name": "name]" }]
}
```

Reply:
Hm, sorry, it might be fixed...

Reply:
- `v-bind` ... https://jsfiddle.net/mysticatea/zuyjom9m/73/
- `v-slot` ... https://jsfiddle.net/mysticatea/zuyjom9m/76/

Looks like `v-slot:[slot.name]` works, but `v-bind` doesn't work. So this is OK.

Reply:
I searched for this issue. I think that it will be fixed by the PR https://github.com/vuejs/vue/pull/9585 :smile: